src/Container/Container.php

```diff
@@ -36,76 +41,131 @@ class Container implements ContainerInterface
      * @param mixed  $value
      * @return Container
      */
-    public function set($key, $value)
+    public function set(string $key, $value): self
     {
+        if (!(is_string($value) || is_object($value))) {
+            throw new \InvalidArgumentException('Value must be a class name, an object instance, or a callable');
+        }
+
         $this->dicValues[$key] = $value;
         unset($this->dicObjects[$key]); // In case an object instance was stored for sharing
 
         return $this;
     }
 
+    public function unset(string $key): void
+    {
+        unset(
+            $this->dicValues[$key],
+            $this->dicObjects[$key]
+        );
+    }
+
+    public function alias(string $alias, string $key): self
+    {
+        $this->aliases[$alias] = $key;
+
+        return $this;
+    }
+
+    public function unalias(string $alias): void
+    {
+        unset($this->aliases[$alias]);
+    }
+
     /**
      * Check if a DIC value/object exists.
      *
      * @param string $key
      * @return bool
      */
-    public function has($key)
+    public function has($key): bool
     {
-        return array_key_exists($key, $this->dicValues);
+        if (isset($this->aliases[$key])) {
+            $key = $this->aliases[$key];
+        }
+
+        return isset($this->dicValues[$key]);
     }
 
     /**
      * @param string $key
      * @return bool
      */
-    public function hasInstance($key)
+    public function hasInstance($key): bool
```

Comment:
```suggestion
    public function hasInstance(string $key): bool
```

and phpdoc can be removed